### ISSUE TITLE
Bug: Scope manager: Identifying `"use strict"` directives is too liberal

### ISSUE THREAD

**Issue by overlookmotel** [typescript-eslint/typescript-eslint#11993]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

scope-manager

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&sourceType=script&showAST=scope&fileType=.js&code=GYVwdgxgLglg9mABMOcAUBKRBvRaDkIAzgKaJFQBOM0%2BGA3IlABYxEB0JAbgIYA2jAL4AoIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
An ESlint test case for `no-eval` rule:


// script
function foo() { ('use strict'); this.eval; }
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

Scope for function `foo` should have `isStrict: false`. `('use strict')` is not a directive.

### Actual Result

Scope for function `foo` has `isStrict: true`.

The above example produces an error with `no-eval` rule with ESLint's built-in scope manager (correctly) but does not with TS-ESLint's scope manager (incorrect). [ESLint playground](https://eslint.org/play/#eyJ0ZXh0IjoiZnVuY3Rpb24gZm9vKCkgeyAoJ3VzZSBzdHJpY3QnKTsgdGhpcy5ldmFsOyB9XG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLWV2YWwiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQifX19)

This same problem also causes 3 tests to fail for `no-invalid-this` rule.

### Additional Info

I believe the cause is that checking for `"use strict"` directives is overly liberal:

https://github.com/typescript-eslint/typescript-eslint/blob/aaa7ca2d6afd8def8cf9d01d0d9deedd38ae1495/packages/scope-manager/src/scope/ScopeBase.ts#L88-L108

It accepts `Literal` nodes without `directive` property defined.

IMO it should only check the `directive` field:

```js
for (const stmt of body.body) {
  if (stmt.type !== AST_NODE_TYPES.ExpressionStatement) {
    break;
  }

  if (stmt.directive == null) {
    break;
  }

  if (stmt.directive === 'use strict') {
    return true;
  }
}
```

This code dates back to the original introduction of `scope-manager` package (#1939). Perhaps the more liberal check was there for some historical reason e.g.

1. Copied from `eslint-scope`, but `eslint-scope` has been fixed since?
2. A workaround for compatibility with another parser which didn't correctly produce `directive` property on `ExpressionStatement`?

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/scope-manager`     | `8.53.1` |

**Comment by bradzacher** [comment]
Looks like it was changed in `eslint-scope` in https://github.com/eslint/eslint-scope/pull/118 (2024, 4 years after we forked the package).

There's probably a bunch of similar small tweaks that they made since we forked that we haven't picked up.

Most of them are probably going to be super rare, 1-in-a-trillion edge case fixes though.
Happy for this to be fixed.

**Comment by overlookmotel** [comment]
PR: #11995.
